### PR TITLE
Feature/login Python 3 content

### PR DESCRIPTION
**Pull request by untra**
The following has python 3 changes that work and coooperate with out models and ldap / handlers . 

```
sudo apt-get install python3
```

Serious changes to the models, basemodel etc.

_questions to ask:_ 
- Should basemodel / driver be combined into one element? They are sharing some simmilar code / responsibilities, but they also have some distinctly different responsibilities

:sweet_potato:

### CODE REVIEW

File: src/handlers/register/culdap_register_handler.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+    def registerUser(self,data):
+        data['date_registered'] = time.strftime('%a %b %d %H:%M:%S %Z %Y')
+        verified = User().verify(data)
+        if len(verified) != 0:
+            logging.error('User: verification errors!')
+            logging.error(verified)
+            return self.verifyCULdapRegistrationPage(username, verified)
+        user_id = User().create_item(data)
+        # user_id = yield r.db(BaseModel.DB).table("User").insert(data).run(BaseModel.conn)
```
Comment (sungbae) [inline src/handlers/register/culdap_register_handler.py:82]: Are you going to keep this or should we just delete this line?

File: src/models/basemodel.py [hunk 2]
```diff
@@ -85,13 +89,13 @@ def check_data(data, fields, required_fields=[]):
         for field in required_fields:
             if field not in data:
                 yield (field, 'Missing field: {}'.format(field))
```
Comment (sungbae) [inline src/models/basemodel.py:91]: I thought we wanted to remove yield everywhere?

File: src/handlers/register/culdap_register_handler.py (hunk 1 above)
Comment (untra) [inline src/handlers/register/culdap_register_handler.py:82]: What line? 82? Keep that, I'm still working on registration. All thats left to be done is give the user a high five for registering and kick them back to index / wherever they were browsing.

File: src/models/basemodel.py (hunk 2 above)
Comment (untra) [inline src/models/basemodel.py:91]: keep it. see: http://stackoverflow.com/questions/231767/what-does-the-yield-keyword-do-in-python

**Comment by untra** [comment]
Press the button before you do anymore backend code!

**Comment by untra** [comment]
get ppssyycchheedd :crocodile:

**Pull request merged**